src/main/java/com/salesforce/dataloader/action/visitor/BulkV2Connection.java

```diff
@@ -0,0 +1,364 @@
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package com.salesforce.dataloader.action.visitor;
+
+import java.io.BufferedReader;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.io.Serializable;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.TimeZone;
+import java.util.zip.GZIPInputStream;
+
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.SSLContext;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.salesforce.dataloader.config.Config;
+import com.sforce.async.AsyncApiException;
+import com.sforce.async.AsyncExceptionCode;
+import com.sforce.async.AsyncXmlOutputStream;
+import com.sforce.async.BatchInfo;
+import com.sforce.async.BulkConnection;
+import com.sforce.async.ContentType;
+import com.sforce.async.JobInfo;
+import com.sforce.async.JobStateEnum;
+import com.sforce.ws.ConnectionException;
+import com.sforce.ws.ConnectorConfig;
+import com.sforce.ws.MessageHandler;
+import com.sforce.ws.MessageHandlerWithHeaders;
+import com.sforce.ws.bind.CalendarCodec;
+import com.sforce.ws.parser.PullParserException;
+import com.sforce.ws.parser.XmlInputStream;
+import com.sforce.ws.parser.XmlOutputStream;
+import com.sforce.ws.transport.Transport;
+import com.sforce.ws.util.FileUtil;
+
+public class BulkV2Connection extends BulkConnection {
+    private static final JsonFactory JSON_FACTORY = new JsonFactory(new ObjectMapper());
+    private static final String URI_STEM_QUERY = "query/";
+    private static final String AUTH_HEADER = "Authorization";
+    private static final String AUTH_HEADER_VALUE_PREFIX = "Bearer ";
+    
+    private String authHeaderValue = "";
+    private String queryLocator = "";
+    private String queryResultNumberOfRecords = "";
+
+    public BulkV2Connection(ConnectorConfig connectorConfig) throws AsyncApiException {
+        super(connectorConfig);
+    }
+    
+    public JobInfo createJob(JobInfo job) throws AsyncApiException {
+        return createOrUpdateJob(job, job.getContentType() == null ? ContentType.JSON
```

Comment:
I believe V2 Ingest only supports CSV jobs. Maybe you should simplify the new code here by removing any code that handles other job types.